.github/workflows/node.js-CI.yml

```diff
@@ -0,0 +1,50 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    
+    steps:
+      - uses: actions/checkout@v2
+      - name: npm install and nest build  
+        run: |
+          npm install
```

Comment:
Should we run `npm ci` instead of `install` since it's the command we'll be using in production?

Reply:
Indeed, I always use `npm ci` on a CI/CD platform, nice catch!

Reply:
Good catch @GJordao 
From: 
- https://docs.npmjs.com/cli/v7/commands/npm-ci
- https://stackoverflow.com/questions/52499617/what-is-the-difference-between-npm-install-and-npm-ci

We should use `npm ci`. 
I believe we want an error when `package-lock.json` doesn't exist or is not equivalent to `package.json`.
Also it is faster :+1: 
Will use it then.

Reply:
@MicroAnibal there will always be a `package-lock.json`, everytime a developer runs `npm i` it creates one.
If it was excluded, it will fail and there fore the pipeline break :+1: 

Also, is something was added to the package.json and the package-lock.json was not updated it will fail with an explicit message that package X is missing!

Reply:
I should notice that the Dockerfile currently does not use npm CI, we should probably change it as well

Reply:
@GJordao, @nneves - Changes Implemented :+1: